### DISCUSSION TITLE
Restrictions on who can run `vercel --prod`

### DISCUSSION THREAD

**Discussion by carsonjones** [vercel/vercel#5493] [Help] [answered]
Hello! I am curious if there any controls one can put into place to limit which users in a team have access to running the `vercel --prod` command?

I've checked the [Deployments](https://vercel.com/docs/platform/deployments) documentation, dug through project settings (security and advanced), and looked at the team level settings (members, general) but I'm not having much luck. Does this kind of control exist today?

We have a use case where we are considering limiting who has "keys for launching to prod", so I'm interested in hearing if this is possible.

🙏 Thanks in advance!

**Comment by styfle** [comment, marked as answer]
Hello Carson,

At the moment, that is not possible. We will release more roles soon so you can have more control over your Vercel deployments and related resources.

If your team decides to use Git deployments (instead of the CLI), permissions can be configured with your Git provider. For example, preventing pushes to the main branch would prevent deployments to prod and setting a required number of reviewers would ensure PRs are reviewed before being merged to the main branch.

**Reply by carsonjones** [reply]
Thanks @styfle ! Very helpful.
